File: 品味Java的21種設計模式/dp_cpp/command/example6/ChopCommand.cpp
#include "ChopCommand.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace command
			{
				namespace example6
				{

					void ChopCommand::setCookApi(CookApi *cookApi)
					{
						this->cookApi = cookApi;
					}

					ChopCommand::ChopCommand(int tableNum)
					{
						this->tableNum = tableNum;
					}

					int ChopCommand::getTableNum()
					{
						return this->tableNum;
					}

					void ChopCommand::execute()
					{
						this->cookApi->cook(tableNum,"绿豆排骨煲");
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/command/example7/ChopCommand.java
package cn.javass.dp.command.example7;

import java.io.Serializable;


/**
 * 命令对象，绿豆排骨煲
 */
public class ChopCommand implements Command, Serializable {
    /**
     * 持有具体做菜的厨师的对象
     */
    private CookApi cookApi = null;
    /**
     * 设置具体做菜的厨师的对象
     * @param cookApi 具体做菜的厨师的对象
     */
    public void setCookApi(CookApi cookApi) {
        this.cookApi = cookApi;
    }
    /**
     * 点菜的桌号
     */
    private int tableNum;
    /**
     * 构造方法，传入点菜的桌号
     * @param tableNum 点菜的桌号
     */
    public ChopCommand(int tableNum) {
        this.tableNum = tableNum;
    }
    public int getTableNum() {
        return this.tableNum;
    }

    public void execute() {
        this.cookApi.cook(tableNum, "绿豆排骨煲");
    }
}
